### ISSUE TITLE
Possible no-loss cap mismatch in capital gains basis indexation scenarios

### ISSUE THREAD

**Issue by MaxGhenis** [Budget-Lab-Yale/Tax-Simulator#126]
## Summary

I noticed a possible mismatch between the public capital gains basis indexation scenario configuration and the published description of the policy.

The [public Yale Budget Lab article for capital gains basis indexation](https://budgetlab.yale.edu/research/indexing-capital-gains-inflation) describes the policy as disallowing indexation from generating losses. However, in the public [`indexed_kg.csv` runscript scenarios at commit `18a25c8f02704c416cb1c30b55a6575b3bf14372`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/runscripts/tests/indexed_kg.csv#L1-L4), the scenarios used for the prospective and retrospective estimates appear to inherit `index_kg_no_loss = 0`, which allows indexation to turn gains into losses.

## Evidence

Commit checked: [`18a25c8f02704c416cb1c30b55a6575b3bf14372`](https://github.com/Budget-Lab-Yale/Tax-Simulator/commit/18a25c8f02704c416cb1c30b55a6575b3bf14372) (`Merge pull request #118 from Budget-Lab-Yale/indexed-kg`).

The runscript uses these two indexation scenarios:

- [`config/runscripts/tests/indexed_kg.csv`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/runscripts/tests/indexed_kg.csv#L1-L4)
  - [`indexed_kg_prospective`, tax law `tests/indexed_kg_prospective`, behavior `kg/62`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/runscripts/tests/indexed_kg.csv#L3)
  - [`indexed_kg_retrospective`, tax law `tests/indexed_kg_retrospective`, behavior `kg/62`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/runscripts/tests/indexed_kg.csv#L4)

Those scenario configs set `index_kg`, `index_kg_min_years_held`, and `index_kg_purchased_after`, but do not set `index_kg_no_loss`:

- [`config/scenarios/tax_law/tests/indexed_kg_prospective/pref.yaml`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/tests/indexed_kg_prospective/pref.yaml#L1-L11)
- [`config/scenarios/tax_law/tests/indexed_kg_retrospective/pref.yaml`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/tests/indexed_kg_retrospective/pref.yaml#L1-L11)

The [baseline default](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/baseline/pref.yaml#L74-L76) is:

```yaml
index_kg_no_loss:
  value: 0
```

with the [in-file comment](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/baseline/pref.yaml#L74):

```yaml
# Whether indexation can turn a gain into a loss (0 = losses allowed, 1 = clamp at zero)
```

The calculator only applies the no-loss cap when this flag is `1`, in [`src/calc/functions/income/kg.R`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/src/calc/functions/income/kg.R#L63-L66):

```r
kg_lt_infl_adj = if_else(pref.index_kg_no_loss == 1 & kg_lt > 0,
                         pmin(kg_lt_infl_adj, kg_lt),
                         kg_lt_infl_adj)
```

There is a separate scenario, [`config/scenarios/tax_law/tests/indexed_kg_retro_noloss/pref.yaml`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/tests/indexed_kg_retro_noloss/pref.yaml#L1-L14), that sets [`index_kg_no_loss: 1`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/scenarios/tax_law/tests/indexed_kg_retro_noloss/pref.yaml#L13-L14), but it does not appear in [`config/runscripts/tests/indexed_kg.csv`](https://github.com/Budget-Lab-Yale/Tax-Simulator/blob/18a25c8f02704c416cb1c30b55a6575b3bf14372/config/runscripts/tests/indexed_kg.csv#L1-L4).

## Why this matters

If the published workbook/results were generated from the public `indexed_kg.csv` runscript, then the estimates appear to allow indexation to create losses, contrary to the policy description. If the workbook/results were generated from a different private runscript or scenario, then the public reproduction path may be stale or incomplete.

There is also a related edge case: even when `index_kg_no_loss = 1`, the current cap only clamps positive `kg_lt` records. Net long-term loss records can still receive an inflation adjustment and become more negative.

## Questions

1. Should `indexed_kg_prospective` and `indexed_kg_retrospective` set `index_kg_no_loss: 1`?
2. Was `indexed_kg_retro_noloss` used for any published table, or only as an unused test scenario?
3. Should the no-loss cap also prevent indexation from increasing net long-term losses on records with `kg_lt < 0`?

Happy to provide more detail if useful.
